lib/rack/content_type_default.rb

```diff
@@ -1,32 +1,62 @@
 # This is a simple Rack Middleware to set the request content type for specific routes. It checks if a content type is
-# set and does not override the existing one. The options allow you to specify the request methods, and the content type
-# to be set.
+# set and does not override the existing one. The options allow you to specify the request methods and the content type
+# to be set. The options also allow you to set a default content type only on select paths. In addition, the options
+# allow  you to set content type to to application/json or application/xml if possible based on the path ending.
 # Copied and adapted from https://gist.github.com/tstachl/6264249
 
+require 'active_support/core_ext'
+
 module Rack
   class ContentTypeDefault
-    def initialize(app, methods = [:post], content_type = 'application/json')
+    def initialize(app, methods = [:post], content_type = 'application/json', paths = 'all', default_based_on_path = false)
       @app = app
       @methods = [*methods].map{ |item| item.to_s.upcase }
       @content_type = content_type
+      @paths = [*paths]
+      @default_based_on_path = default_based_on_path
     end
 
     def call(env)
       req = Rack::Request.new(env)
+      @app.call(env) and return unless reset_content_type?(req)
```

Comment:
So does this end up allowing `@app.call(env)` to be run twice if `reset_content_type` returns false?
```ruby
[1] pry(main)> def low_precendence_and(bool)
[1] pry(main)*   puts("foo") and return unless bool
[1] pry(main)*   "bar"
[1] pry(main)* end
=> :low_precedence_and
[2] pry(main)> low_precedence_and(true)
=> "bar"
[3] pry(main)> low_precedence_and(false)
foo
=> "bar"
```